google/gax/api_callable.py

```diff
@@ -254,29 +251,55 @@ def _construct_retry(method_config, retry_codes, retry_params, retry_names):
     Returns:
       A RetryOptions object, or None.
     """
-    if not method_config:
+    if method_config is None:
         return None
-    codes = []
-    if retry_codes:
-        for codes_name in retry_codes:
-            if (codes_name == method_config['retry_codes_name'] and
-                    retry_codes[codes_name]):
-                codes = [
-                    retry_names[name] for name in retry_codes[codes_name]]
-                break
 
-    params_struct = None
-    if method_config.get('retry_params_name'):
-        for params_name in retry_params:
-            if params_name == method_config['retry_params_name']:
-                params_struct = retry_params[params_name].copy()
-                break
-        backoff_settings = BackoffSettings(**params_struct)
-    else:
-        backoff_settings = None
+    codes = None
+    if retry_codes and 'retry_codes_name' in method_config:
+        codes_name = method_config['retry_codes_name']
+        if codes_name in retry_codes and retry_codes[codes_name]:
+            codes = [retry_names[name] for name in retry_codes[codes_name]]
+        else:
+            codes = []
+
+    backoff_settings = None
+    if retry_params and 'retry_params_name' in method_config:
+        params_name = method_config['retry_params_name']
+        if params_name and params_name in retry_params:
+            backoff_settings = BackoffSettings(**retry_params[params_name])
+
+    return RetryOptions(retry_codes=codes, backoff_settings=backoff_settings)
+
+
+def _merge_retry_options(retry, overrides):
+    """Helper fo ``construct_settings()``.
+
+    This takes two retry options, and merges them into a single RetryOption
+    instance.
+
+    Args:
+      retry: The base RetryOptions.
+      overrides: The RetryOptions used for overriding ``retry``. Use the values
+        if it is not None. If entire ``overrides`` is None, that means cancel of
```

Comment:
"that means cancel of specifying retry options, thus returns None" ---> "ignore the base retry options and return None"